### PR TITLE
Sync main and latest release

### PR DESCRIPTION
**Pull request by alex-ioja-yang-cko**
Following [4.0.5 release](https://github.com/checkout/frames-ios/releases/tag/4.0.5) we need to synchronise changes back to main branch.

PR contains many files & lines changed in `docs` folder, but this is irrelevant to the overall PR and is created automatically. Ignore that folder to avoid distractions 😄

### CODE REVIEW

**Review by deepesh-vasthimal-cko — COMMENTED** [review]
Looks good overall. Please have a look at the few comments from @ehab-al-cko

File: README.md [hunk 1]
```diff
@@ -72,6 +72,8 @@ source 'https://github.com/CocoaPods/Specs.git'
 platform :ios, '12.0'
 use_frameworks!
 
+pod 'PhoneNumberKit', :git => 'https://github.com/marmelroy/PhoneNumberKit'
```
[the remark this answers is not in the document]
Comment (alex-ioja-yang-cko) [reply README.md:75]: We have had a chat over this at one of the standups. For whatever reason, PhoneNumberKit is no longer publishing their versions to Cocoapods. The last version available in the Pod Registry is 3.3.4, which is fairly old and apparently with some formatting bugs.

In order to get anything more recent than 3.3.4 (it currently stands at 3.5.10) the application needs to provide the source url for the dependency. This has also been added to our external docs and is an important detail of consuming Frames via CocoaPods

File: Source/Resources/en.lproj/Localizable.strings [hunk 2]
```diff
@@ -4,7 +4,7 @@
 "missingBillingFormCity" = "Missing City";
 "missingBillingFormCounty" = "Missing County";
 "missingBillingFormPostcode" = "Missing Postcode";
-"missingBillingFormPhoneNumber" = "Please enter a valid phone number";
+"missingBillingFormPhoneNumber" = "Please enter a valid phone number, with a valid international prefix";
```
[the remark this answers is not in the document]
Comment (alex-ioja-yang-cko) [reply Source/Resources/en.lproj/Localizable.strings:7]: This has already been confirmed. Patrick has accepted we will only make this update for English.

File: Source/Core/Constants/AccessibilityIdentifiers.swift [hunk 3]
```diff
@@ -0,0 +1,29 @@
+//
+//  AccessibilityIdentifiers.swift
+//  
+//
+//  Created by Alex Ioja-Yang on 04/05/2023.
+//
+
+import Foundation
+
+public enum AccessibilityIdentifiers {
```
[the remark this answers is not in the document]
Comment (alex-ioja-yang-cko) [reply Source/Core/Constants/AccessibilityIdentifiers.swift:10]: I do not think descriptions are required. This is not production code, at most may be used in UITesting. Even there, I'm not certain they can be accessed directly.

File: Source/UI/BillingForm/BillingFormCell.swift [hunk 4]
```diff
@@ -51,6 +51,20 @@ import Foundation
...
+    var accessibilityIdentifier: String {
+        typealias Identifier = AccessibilityIdentifiers.BillingForm
+        switch self {
+        case .fullName: return Identifier.cardholder
+        case .addressLine1: return Identifier.addressLine1
+        case .addressLine2: return Identifier.addressLine2
+        case .city: return Identifier.city
+        case .state: return Identifier.state
+        case .postcode: return Identifier.postcode
+        case .country: return ""
```
[the remark this answers is not in the document]
Comment (alex-ioja-yang-cko) [reply Source/UI/BillingForm/BillingFormCell.swift:63]: Country is a button and when writing UITests its not found under the same format, its instead using the localised button title. It is therefore redundant to create an AccessibilityLabel enum value, and without an AccessibilityLabel enum value there is no `accessibilityIdentifier` to assign. We do not commonly add comments for why a certain value in a switch statement is returned, and given there is no complex business logic attached, I do not see a reason to start here 🤔

File: Tests/UI/BillingForm/View/BillingFormTextFieldViewTests.swift [hunk 5]
```diff
@@ -24,11 +24,6 @@ class BillingFormTextFieldViewTests: XCTestCase {
 
     func testTextFieldContainerStyle() {
         XCTAssertEqual(view.textFieldContainer.backgroundColor, style.textfield.backgroundColor)
-        XCTAssertEqual(view.textFieldContainer.layer.borderColor, style.textfield.normalBorderColor.cgColor)
-        XCTAssertEqual(view.textFieldContainer.layer.cornerRadius, style.textfield.cornerRadius)
-        XCTAssertEqual(view.textFieldContainer.layer.borderWidth, style.textfield.borderWidth)
-        XCTAssertEqual(view.textFieldContainer.layer.cornerRadius, style.textfield.cornerRadius)
-        XCTAssertEqual(view.textFieldContainer.layer.borderWidth, style.textfield.borderWidth)
```
[the remark this answers is not in the document]
Comment (alex-ioja-yang-cko) [reply Tests/UI/BillingForm/View/BillingFormTextFieldViewTests.swift]: These tests are not in `main`, and opening the Files Changed tab they do not appear as a change, so I am unsure how you created this comment

File: Source/UI/BillingForm/View/BillingFormCellTextField.swift [hunk 6]
```diff
@@ -32,6 +32,7 @@ final class BillingFormCellTextField: UITableViewCell {
         self.type = type
         self.style = style
         self.tag = tag
+        textFieldView?.accessibilityIdentifier = type?.accessibilityIdentifier
         textFieldView?.update(style: style, type: type, textFieldValue: textFieldValue, tag: tag)
```
[the remark this answers is not in the document]
Comment (alex-ioja-yang-cko) [reply Source/UI/BillingForm/View/BillingFormCellTextField.swift]: This was getting assigned twice, updated 👍

File: Source/UI/CommonUI/View/Component/InputView.swift [hunk 7]
```diff
@@ -40,7 +40,7 @@ class InputView: UIView {
         BorderView().disabledAutoresizingIntoConstraints()
     }()
 
-    private(set) lazy var textFieldContainer: UIStackView = {
+    private lazy var textFieldContainer: UIStackView = {
```
[the remark this answers is not in the document]
Comment (alex-ioja-yang-cko) [reply Source/UI/CommonUI/View/Component/InputView.swift]: This is not a change being commit, `main` already has this property private

File: README.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (alex-ioja-yang-cko) [reply README.md:75]: Our ReadMe should be on point and guide to achieve outcome. I don't think this is the place to offer extensive context over why or how. We're also not in control of PhoneNumberKit, so simply documenting the integration steps in the ReadMe will achieve outcome whilst avoiding noise.

File: Source/Core/Constants/AccessibilityIdentifiers.swift (hunk 3 above)
[the remark this answers is not in the document]
Comment (alex-ioja-yang-cko) [reply Source/Core/Constants/AccessibilityIdentifiers.swift:11]: I am not 100% certain whether the end consumer can/cannot actually interact with these values directly, hence I would for now prefer to try and keep public. Even if they can't use IDE to link to them, if we say "Use this in your tests", at least they will be confident this is meant to be used publicly.

I have therefore added descriptions to the elements 👍

File: Source/UI/BillingForm/BillingFormCell.swift (hunk 4 above)
[the remark this answers is not in the document]
Comment (alex-ioja-yang-cko) [reply Source/UI/BillingForm/BillingFormCell.swift:65]: Added comment 👍

File: README.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (alex-ioja-yang-cko) [reply README.md:77]: I did now update to include a comment as suggested

**Pull request merged**